gateway/gateway_test.go

```diff
@@ -144,23 +152,31 @@ var _ = Describe("Gateway", func() {
 		var clearDB = false
 
 		It("should wait for backend config", func() {
-			gateway.Setup(c.mockBackendConfig, c.mockJobsDB, nil, c.mockStats, &clearDB)
+			gateway.Setup(c.mockApp, c.mockBackendConfig, c.mockJobsDB, nil, c.mockStats, &clearDB)
 		})
 	})
 
 	Context("Valid requests", func() {
 		var (
-			gateway      = &HandleT{}
-			clearDB bool = false
+			gateway                = &HandleT{}
+			clearDB           bool = false
+			gatewayBatchCalls int  = 1
 		)
 
+		// tracks expected batch_id
+		nextBatchID := func() (batchID int) {
```

Comment:
Why do you need a separate function for this?

Reply:
Its just a utility in order to not have to remember to increate gatewayBatchCalls every time a batch id is needed.